library/alloc/src/fmt.rs

```diff
@@ -354,7 +354,7 @@
 //! sign := '+' | '-'
 //! width := count
 //! precision := count | '*'
-//! type := '?' | 'x?' | 'X?' | identifier
+//! type := '?' | 'x' | 'X' | 'o' | 'b' | 'e' | 'E' | 'p'
```

Comment:
```suggestion
//! type := '?' | 'x' | 'X' | 'x?' | 'X?' | 'o' | 'b' | 'e' | 'E' | 'p'
```
`x?` is allowed and means “`Debug`, but prints numbers in hex”:
```rs
fn main() {
    println!("{:?}", 12); // 12
    println!("{:x?}", 12); // c
}
```